go/merge/three_way_list.go

```diff
@@ -11,6 +11,53 @@ import (
 	"github.com/attic-labs/noms/go/types"
 )
 
+// Merge rules for List are a bit more complex than Map, Struct, and Set due
```

Comment:
I think we should just put the list merge algorithm in three_way.go so that we can link to the entire merge algorithm in one place in the documentation.